CMakeLists.txt

```diff
@@ -14,6 +14,15 @@ if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE "${DEFAULT_BUILD_TYPE}")
 endif()
 
+# Check whther 3rd-party modules exist
+foreach(MODULE IN ITEMS antlr/antlr4 cppitertools cuda-samples isl pybind11 z3)
+    file(GLOB RESULT ${CMAKE_CURRENT_SOURCE_DIR}/3rd-party/${MODULE}/*)
+    list(LENGTH RESULT RES_LEN)
+    if(RES_LEN EQUAL 0)
+        message(FATAL_ERROR "3rd-party module ${MODULE} not found, please clone with `git clone --recursive <path-to-this-repo>`")
```

Comment:
Given the repo is already cloned, maybe `git submodule update --init --recursive` would be better?